asapdiscovery-dataviz/asapdiscovery/dataviz/html_viz.py

```diff
@@ -128,36 +137,326 @@ def write_html(html, path):
+        save_openeye_pdb(combine_protein_ligand(self.protein, pose), "tmp_complex.pdb")
+
+        # run the PLIP CLI.
+        subprocess.run(["plip", "-f", "tmp_complex.pdb", "-x", "-o", "tmp"])
+
+        # load the XML produced by PLIP that contains all the interaction data.
+        intn_dict_xml = xmltodict.parse(
+            ET.tostring(ET.parse("tmp/report.xml").getroot())
+        )
+
+        intn_dict = {}
+        intn_counter = 0
+        # wrangle all interactions into a dict that can be read directly by 3DMol.
+        for _, data in intn_dict_xml["report"]["bindingsite"]["interactions"].items():
+            if data:
+                for intn_type, intn_data in data.items():
+                    if isinstance(
+                        intn_data, list
+                    ):  # multiple interactions of this type
+                        for intn_data_i in intn_data:
+                            k, v = self.build_interaction_dict(
+                                intn_data_i, intn_counter, intn_type
+                            )
+                            intn_dict[k] = v
+                            intn_counter += 1
+
+                    elif isinstance(intn_data, dict):  # single interaction of this type
+                        k, v = self.build_interaction_dict(
+                            intn_data, intn_counter, intn_type
+                        )
+                        intn_dict[k] = v
+                        intn_counter += 1
+
+        # remove tmp files/folder.
+        os.remove("tmp_complex.pdb")
+        shutil.rmtree("tmp")
+
+        return intn_dict
+
+    def get_html_airium(self, pose):
+        """
+        Get HTML for visualizing a single pose. This uses Airium which is a handy tool to write
+        HTML using python. We can't do f-string because of all the JS curly brackets, need to do '+' instead.
+        """
+        a = Airium()
+
+        # first prep the coloring function.
+        surface_coloring = self.get_color_dict()
+        residue_coloring_function_js = ""
+        start = True
+        for color, residues in surface_coloring.items():
+            residues = [str(res) for res in residues]
+            if start:
+                residue_coloring_function_js += (
+                    "if (["
+                    + ",".join(residues)
+                    + "].includes(atom.resi)){ \n return '"
+                    + color
+                    + "' \n "
+                )
+                start = False
+            else:
+                residue_coloring_function_js += (
+                    "} else if (["
+                    + ",".join(residues)
+                    + "].includes(atom.resi)){ \n return '"
+                    + color
+                    + "' \n "
+                )
+
+        # start writing the HTML doc.
+        a("<!DOCTYPE HTML>")
+        with a.html(lang="en"):
+            with a.head():
+                a.meta(charset="utf-8")
```

Comment:
Think I agree with @chrisiacovella that using `jinja` plus a template may be less painful than building up the script below. I think you could also use a combination if you prefer, using `airium` to build up the overall HTML doc but using `jinja` to populate the JS script.

Reply:
I agree. I'll discuss with @hmacdope and @mikemhenry before starting a PR to introduce `jinja` throughout the codebase.